src/functionalTest/java/uk/gov/hmcts/reform/wataskmanagementapi/controllers/DeleteTaskByIdControllerTest.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.hmcts.reform.wataskmanagementapi.controllers;
+
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.http.HttpStatus;
+import uk.gov.hmcts.reform.wataskmanagementapi.SpringBootFunctionalBaseTest;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.InitiateTaskRequest;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.TerminateTaskRequest;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.entities.TaskAttribute;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TerminateReason;
+import uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.options.TerminateInfo;
+import uk.gov.hmcts.reform.wataskmanagementapi.domain.entities.TestVariables;
+
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singleton;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.InitiateTaskOperation.INITIATION;
+import static uk.gov.hmcts.reform.wataskmanagementapi.controllers.request.enums.TaskAttributeDefinition.TASK_CASE_ID;
+
+public class DeleteTaskByIdControllerTest extends SpringBootFunctionalBaseTest {
+
+    private static final String ENDPOINT_BEING_TESTED = "task/{task-id}";
+    private Headers authenticationHeaders;
+
+    @Before
+    public void setUp() {
+        //Reset role assignments
+        authenticationHeaders = authorizationHeadersProvider.getTribunalCaseworkerAAuthorization();
+        common.clearAllRoleAssignments(authenticationHeaders);
+    }
+
+    @Test
+    public void should_succeed() {
```

Comment:
Even though under the hood the COMPLETE & CANCEL behaviour is identical, should we still include a happy path for createTaskAndCancel?

Reply:
in this scenario they are used to prepare the data the logic should be covered as part of the other classes you mentioned

Reply:
Added extra test functional and integration to cover the discussed journey